packages/graphql/src/schema/validation/custom-rules/directive-argument-of-correct-type.ts

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright (c) "Neo4j"
+ * Neo4j Sweden AB [http://neo4j.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import type { ASTVisitor, DirectiveNode, DocumentNode, GraphQLArgument, ArgumentNode } from "graphql";
+import { GraphQLError, coerceInputValue, valueFromASTUntyped, buildASTSchema, Kind } from "graphql";
+
+import type { SDLValidationContext } from "graphql/validation/ValidationContext";
+
+export function DirectiveArgumentOfCorrectType(context: SDLValidationContext): ASTVisitor {
+    const astDefinitions = context.getDocument().definitions;
+
+    const schemaDocument: DocumentNode = {
+        kind: Kind.DOCUMENT,
+        definitions: astDefinitions,
+    };
+    const schema = buildASTSchema(schemaDocument, { assumeValid: true, assumeValidSDL: true });
+
+    return {
+        Directive(directiveNode: DirectiveNode) {
+            // Validate only Authorization usage
+            if (!directiveNode.name.value.includes("Authorization")) {
+                return;
+            }
+
+            const graphQLDirective = schema.getDirective(directiveNode.name.value);
```

Comment:
what do you think about naming this variable something like `directiveDefinition`? So it's consistent with the `argumentDefinition` on line 49.
Also the `directiveNode` could be `directiveUsage` so it's clearer what it's referring to? It already has the type `DirectiveNode`..

Reply:
I like it!